classes/product.js

```diff
@@ -0,0 +1,52 @@
+class Product
+{
+    /**
+     * @constructor Represents a product.
+     *
+     * @param {string} name - The display name for this product.
+     * @param {int} quantity - The amount of this product
+     * @param {float} size - A unitless measure for the size of a product (qty = 1).
+     * @param {int=1} perishable - The number of days a perishable good remains fresh. 0 for infinite.
+     */
+    constructor(name, quantity, size, perishable = 0)
+    {
+        this.name = String(name);
+        this.quantity = parseInt(quantity);
+        this.size = parseFloat(size);
```

Comment:
@lukko Does this seem reasonable? I think this is pretty much what you suggested :).